pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -607,6 +618,94 @@ func (ctrl *Controller) mergeConfigChanges(origFile *igntypes.File, cfg *mcfgv1.
 	return cfgTOML, ctrl.syncStatusOnly(cfg, nil)
 }
 
+// createCapabilitiesMC creates a Machine Config for the crio drop-in capabilities file.
+// We are dropping the NET_RAW and SYS_CHROOT capabilitie by default and in order to not break
+// existing clusters, we are creating a drop-in file at /etc/crio/crio.conf.d that has NET_RAW
+// and SYS_CHROOT as part of the default capabilities list. Users then have the option to delete the
+// capabilities MC that creates this drop-in file.
+// Create a crio-capabilities config map to keep track of whether this capabilities MC was already created
+// once. This ensures that if the user deletes the capabilities MC, the controller doesn't recreate it on a
+// resync.
+func (ctrl *Controller) createCapabilitiesMC() error {
+	var configMapName = "crio-capabilities"
+
+	// Check if the crio-capabilities config map exists in the openshift-machine-config-operator namespace
+	capCM, err := ctrl.kubeClient.CoreV1().ConfigMaps(ctrl.namespace).Get(context.TODO(), configMapName, metav1.GetOptions{})
+	capCMIsNotFound := errors.IsNotFound(err)
+	if err != nil && !capCMIsNotFound {
+		return fmt.Errorf("error checking for %s config map: %v", configMapName, err)
+	}
+	// If the crio-capabilities config map exists, that means the capabilities MC was already created
+	// so we should not create this MC again and return
+	if capCM != nil && !capCMIsNotFound {
+		return nil
+	}
+	// Create the config map for capabilities so we know that the capabilities MC has been created
+	capCM.Name = configMapName
+	capCM.Namespace = ctrl.namespace
+	if _, err := ctrl.kubeClient.CoreV1().ConfigMaps(ctrl.namespace).Create(context.TODO(), capCM, metav1.CreateOptions{}); err != nil {
+		return fmt.Errorf("error creating %s config map: %v", configMapName, err)
+	}
+
+	sel, err := metav1.LabelSelectorAsSelector(metav1.AddLabelToSelector(&metav1.LabelSelector{}, builtInLabelKey, ""))
+	if err != nil {
+		return err
+	}
+	// Find all the MachineConfigPools
+	mcpPoolsAll, err := ctrl.mcpLister.List(sel)
+	if err != nil {
+		return err
+	}
+
+	// Create the capabilities MC for all the available pools
+	for _, pool := range mcpPoolsAll {
```

Comment:
Hmm, today all custom pools must inherit from the `worker` pool. By looping through all pools aren't you effectively double-adding capabilities as a drop-in to a custom pool twice? Once as a worker MC and once as a custom pool MC.

Maybe we should just add it to master/worker.

Reply:
Oh I just tested this and it seems to only apply select master/worker even with a custom pool, maybe this isn't an issue

Reply:
We want to apply the capabilities MC to all the available pools in the cluster. Some customers have more than master and worker pool and I thought using the `LabelSelectorAsSelector` function gets you all the available pools in the cluster. I might be wrong, but so far I have only tested it on clusters that have master and worker pools only.

Reply:
Yeah I created a cluster, added an `infra` pool, and upgraded to this PR, and it only created capabilities MC for master/worker.

Today in the MCO we only allow one custom pool, and it MUST inherit from worker if I recall correctly. So in essence in this case the `infra` node would have the capabilities MC through the worker MC, so I guess in that sense it will work. Although I am not very sure why selecting the way you did didnt generate a infra MC. I'll take a look but it sounds like the behaviour is what we want anyways?

Reply:
Does this mean that if the user deletes the worker capabilities MC, the changes will roll out to both worker and infra nodes? I think we want each pool to have its own capabilities MC so that users can delete the MC for the specific pools where they want to drop these capabilities.
And is this behaviour only for the infra pool, or it will be for all other custom pools the user creates? If this is the way it is done now, we could probably document that deleting the worker pool capabilities MC will drop these capabilities on nodes that are in the custom pools as well as the worker pool.